### ISSUE TITLE
Some community servers have weird time in `status`

### ISSUE THREAD

**Issue by Bash-09** [MegaAntiCheat/client-backend#15]
On one community server I play on they hide the time connected from the status command, in a couple cases this resulted in weird invalid values, "06:06:" and "1:s". Need to figure out what to do about that, perhaps make the regex resilient to these weird time values (maybe replace with a default if it doesn't parse).

![image](https://github.com/MegaAntiCheat/client-backend/assets/47521168/ee1b3f3d-5dde-47b4-9c48-ced27cae99b8)

**Comment by Bash-09** [comment]
TODO:
* Update regex capture group for time to something like `(\S+)`
* Change type for time in player object to `String` or `Arc<str>`
* Update API docs to include time as a String instead of an int

**Comment by THE-EXP** [comment]
You could instead process it line-by-line splitting each line into an array(and filter it using Itertools's .Iter().unique() to remove all the duplicate empty strings from the iterator(bc the space will be consumed by .split() and the empty string is not of any use)
and then pass the elements into Player construct where you can then parse the time separately from everything else and turn it into seconds or just format it properly into a string
I know that might not be as performant because .iter().unique() will need to make a duplicate of the data, but it's all i can think of for now
(i am in no way a Rust pro or even a beginner, but i hope i can at least help you guys with something, the project is ambitious and i honestly all in on it where possible)

**Comment by lili7h** [comment]
So currently the way we read status is by eternally watching the output of `console.log` and running each new line through a regex. Currently the regex that recognises player entries in the `status` output has a given capture group for time, which occasionally will fail to match on the fucky wucky times that community servers give the players. The simplest but somewhat spaghetti way to deal with this would be to rework the player regex to try and 'fuzzy match' the time value.

Here is the current player regex:
```regex
^#\s*(\d+)\s"(.*)"\s+(\[U:\d:\d+\])\s+(\d*:?\d\d:\d\d)\s+(\d+)\s*(\d+)\s*(\w+).*$
```

And that would match any of the following lines (but only 1 at a time):
```
#    376 "ChunionFragz"      [U:1:123858189]     37:35       74    0 active
#    408 "THE FLYING HEAVY"  [U:1:87983792]      02:04       70    0 active
#    303 "CannedCat"         [U:1:1024285746]     1:26:58    70    0 active
#    399 "Bloodmasked"       [U:1:80378183]      17:18       70    0 active
#    416 "ox"                [U:1:112116659]     00:22       72    0 active
#    411 "[TopG] Kam"        [U:1:1536942125]    01:49       72    0 active
#    412 "Furry Lover 2000"  [U:1:1219272849]    01:49       97    0 active
#    420 "solderweed2121"    [U:1:1467853567]    00:13      151   51 active
#    413 "thenaturalnonwhite" [U:1:1115521697]   01:14       74    0 active
#    391 "Josef Gangl"       [U:1:1180966211]    28:54       65   11 active
#    414 "c1mpleZ_"          [U:1:38367492]      00:28       55    0 active
#    415 "PivotLime"     `    [U:1:150921345]     00:26       70   65 spawning
#    392 "♡shiftyvans♡"  [U:1:838717415]     28:20       60    0 active
#    417 "Lilith"            [U:1:111216987]     00:21       58    0 active
#    356 "Greg Katani"       [U:1:41954354]      59:35       67    0 active
#    418 "kreme"             [U:1:116156325]     00:17       58    0 active
#    405 "sapperman2"        [U:1:124685973]     10:45       82    0 active
#    395 "Miss Swaggy"       [U:1:1284835482]    22:58       71    0 active
#    419 "The man himself"   [U:1:1550661956]    00:17      136    7 active
#    407 "a e r g h"         [U:1:49946951]      03:00       78    0 active
#    394 "Sully"             [U:1:183194331]     27:12      101    0 active
``` 

This is an example from a matchmaking server, so these are the normal expected time values.

If possible, we would greatly prefer to keep the time as a capture group as well.

![image](https://github.com/MegaAntiCheat/client-backend/assets/85176789/48096f30-40e5-467d-a281-c339f58a46a2)

**Comment by THE-EXP** [comment]
I have played around with the regex from previous message, modified the test data from:
```text/plain
#    376 "ChunionFragz"      [U:1:123858189]     37:35       74    0 active
#    408 "THE FLYING HEAVY"  [U:1:87983792]      02:04       70    0 active
#    303 "CannedCat"         [U:1:1024285746]     1:26:58    70    0 active
#    399 "Bloodmasked"       [U:1:80378183]      17:18       70    0 active
#    416 "ox"                [U:1:112116659]     00:22       72    0 active
#    411 "[TopG] Kam"        [U:1:1536942125]    01:49       72    0 active
#    412 "Furry Lover 2000"  [U:1:1219272849]    01:49       97    0 active
#    420 "solderweed2121"    [U:1:1467853567]    00:13      151   51 active
#    413 "thenaturalnonwhite" [U:1:1115521697]   01:14       74    0 active
#    391 "Josef Gangl"       [U:1:1180966211]    28:54       65   11 active
#    414 "c1mpleZ_"          [U:1:38367492]      00:28       55    0 active
#    392 "♡shiftyvans♡"  [U:1:838717415]     28:20       60    0 active
#    417 "Lilith"            [U:1:111216987]     00:21       58    0 active
#    356 "Greg Katani"       [U:1:41954354]      59:35       67    0 active
#    418 "kreme"             [U:1:116156325]     00:17       58    0 active
#    405 "sapperman2"        [U:1:124685973]     10:45       82    0 active
#    395 "Miss Swaggy"       [U:1:1284835482]    22:58       71    0 active
#    419 "The man himself"   [U:1:1550661956]    00:17      136    7 active
#    407 "a e r g h"         [U:1:49946951]      03:00       78    0 active
#    394 "Sully"             [U:1:183194331]     27:12      101    0 active
```
To:
```text/plain
#    376 "ChunionFragz"      [U:1:123858189]     37:35       74    0 active
#    408 "THE FLYING HEAVY"  [U:1:87983792]      02:04       70    0 active
#    303 "CannedCat"         [U:1:1024285746]     1:26:s    70    0 active
#    399 "Bloodmasked"       [U:1:80378183]      17:18       70    0 active
#    416 "ox"                [U:1:112116659]     00:22:      72    0 active
#    411 "[TopG] Kam"        [U:1:1536942125]    10:a       72    0 active
#    412 "Furry Lover 2000"  [U:1:1219272849]    01:49       97    0 active
#    420 "solderweed2121"    [U:1:1467853567]    00:13      151   51 active
#    413 "thenaturalnonwhite" [U:1:1115521697]   01:14       74    0 active
#    391 "Josef Gangl"       [U:1:1180966211]    28:54       65   11 active
#    414 "c1mpleZ_"          [U:1:38367492]      00:28       55    0 active
#    392 "♡shiftyvans♡"  [U:1:838717415]     28:20       60    0 active
#    417 "Lilith"            [U:1:111216987]     00:21       58    0 active
#    356 "Greg Katani"       [U:1:41954354]      59:35       67    0 active
#    418 "kreme"             [U:1:116156325]     00:17       58    0 active
#    405 "sapperman2"        [U:1:124685973]     10:45       82    0 active
#    395 "Miss Swaggy"       [U:1:1284835482]    22:58       71    0 active
#    419 "The man himself"   [U:1:1550661956]    00:17      136    7 active
#    407 "a e r g h"         [U:1:49946951]      03:00       78    0 active
#    394 "Sully"             [U:1:183194331]     27:0s      101    0 active
```

With the first training set and initial regex, every line is captured correctly< however it doesn't capture the lines that are modified in the modified test set, the newly modified regex
```
/^#\s*(\d+)\s"(.*)"\s+(\[U:\d:\d+\])\s+(\d+:?.*?)\s+(\d+)\s*(\d+)\s*(\w+).*$/gm
```

Captures the lines that were modified from the original set, and even parses the time correctly into the group 4 which for the last line will contain the whole "27:0s"

Here are some screenshots from [regexr](https://regexr.com/7ivd5)(link leads to the pattern shown in screenshots below)
![Last line results](https://github.com/MegaAntiCheat/client-backend/assets/105677206/f1e7cde7-79b5-4bf8-a851-ad91f476214d)
![Result for Line 5](https://github.com/MegaAntiCheat/client-backend/assets/105677206/5cfb48da-45d0-4b47-8f5a-18ab066d7005)
![result for Line 3](https://github.com/MegaAntiCheat/client-backend/assets/105677206/8bf5acef-6d3a-4943-bbfe-c113aa089c69)

**Comment by THE-EXP** [comment]
I have fiddled with the regex a little bit more and managed to make it work without any .* or .+ on the modified test set from before

Here's the updated Regex(the link to [regexr](https://regexr.com/7ivd5) stays the same):
```
/^#\s*(\d+)\s"(.*)"\s+(\[U:\d:\d+\])\s+((?:\d|\w+:?)+)\s+(\d+)\s*(\d+)\s*(\w+).*$/gm
```

**Comment by Bash-09** [comment]
Eh I've put a workaround so the regex doesn't reject the weird times, but it'll just fail to parse the time and have it set to 0 instead. This shouldn't really be a problem, just that connections times for some players will be wrong on some community servers (but usually it's because those servers hide connection times anyways, so no loss?).

**Comment by THE-EXP** [comment]
> Eh I've put a workaround so the regex doesn't reject the weird times, but it'll just fail to parse the time and have it set to 0 instead. This shouldn't really be a problem, just that connections times for some players will be wrong on some community servers (but usually it's because those servers hide connection times anyways, so no loss?).

I guess if it works, then it works?
